.github/workflows/compile-check.yml

```diff
@@ -0,0 +1,28 @@
+name: compile-check
```

Comment:
We should probably add some description to the file so that it's clearer what it does. (Should be done to the other CI file as well but that's outside this PR)

Maybe something like

```suggestion
# This action checks whether all Squirrel files compile successfully using standalone Squirrel compiler
name: compile-check
```

Reply:
![image](https://user-images.githubusercontent.com/66967891/217660473-217150f1-cd98-4447-aad0-1ba1edd8a46b.png)
yes github, that is definitely what is happening there